src/main/java/com/thealgorithms/misc/TwoSumProblem.java

```diff
@@ -0,0 +1,32 @@
+package com.thealgorithms.misc;
+
+import java.util.*;
+
+public class TwoSumProblem {
+    /**
+     * The function "twoSum" takes an array of integers and a target integer as input, and returns an
+     * array of two indices where the corresponding elements in the input array add up to the target.
+     * @param nums An array of integers.
+     * @param target The target is the sum that we are trying to find using two numbers from the given array.
+     * @return The method `twoSum` returns an array of integers.
+     * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+     */
+    public static int[] twoSum(int[] arr, int target) {
+        HashMap<Integer, Integer> map = new HashMap<>();
+        for (int i = 0; i < arr.length; i++) {
```

Comment:
Instead of first adding all elements and then check for answer , we can do this in one loop itself by first checking for answer and then adding element in hashmap, this way time taken for program to run will be reduced and extra condition `index == i `will be be removed.

Reply:
This is O(n) time complexity(TC).
What's the TC of your approach?

Reply:
TC will be same O(n) is worst case but time taken will be almost half, as we will run only one loop. I am proposing this way.

```
public static int[] twoSum(int[] arr, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < arr.length; i++) {
            int rem = target - arr[i];
            if (map.containsKey(rem)) {
                int index = map.get(rem);
                return new int[] {index, i};
            }
            if( !map.containsKey(arr[i])) {
                map.put(arr[i], i);
            }
        }
        return new int[] {};
    }
```

Reply:
ohho! yes.